source/common/grpc/grpc_web_filter.cc

```diff
@@ -119,6 +119,11 @@ Http::FilterTrailersStatus GrpcWebFilter::encodeTrailers(Http::HeaderMap& traile
   const uint32_t length = htonl(temp.length());
   buffer.add(&length, 4);
   buffer.move(temp);
+  if (is_text_response_) {
+    std::string encoded = Base64::encode(buffer, buffer.length());
+    buffer.drain(buffer.length());
```

Comment:
Might be clearer to use a second buffer rather than draining here, it shows the logical flow. Overhead is the same, it's all just slices I believe.